monai/transforms/spatial/array.py

```diff
@@ -2609,3 +2619,168 @@ def _get_params(
+                    patch = np.ones((array.shape[0], *self.patch_size)) * self.pad_opts["constant_values"]
+                else:
+                    patch = np.zeros((array.shape[0], *self.patch_size))
+                location = np.zeros((3, len(self.patch_size)))
+                output += [(patch, location)] * (self.fix_num_patches - len(output))
+
+        if not isinstance(array, np.ndarray):
+            output = [convert_to_dst_type(src=patch, dst=array)[0] for patch in output]
+
+        self.num_patches = len(output)
+        return output
+
+
+class RandGridPatch(GridPatch, RandomizableTransform):
+    """
+    Return all (or a subset of) the patches sweeping the entire image with a random starting position.
+
+    Args:
+        patch_size: size of patches to generate slices for, 0 or None selects whole dimension
+        start_pos: starting position in the array, default is 0 for each dimension.
+            np.random.randint(0, patch_size, 2) creates random start between 0 and `patch_size` for a 2D image.
+        fix_num_patches: number of patches to return. Defaults to None, which returns all the available patches.
+        overlap: the amount of overlap of neighboring patches in each dimension (a value between 0.0 and 1.0).
+            If only one float number is given, it will be applied to all dimensions. Defaults to 0.0.
+        sort_key: a callable or string that defines the order of the patches to be returned. If it is a callable, it
+            will be passed directly to the `key` argument of `sorted` function. The string can be "min" or "max",
+            which are, respectively, the minimum and maximum of the sum of intensities of a patch across all dimensions
+            and channels. Also "random" creates a random order of patches.
+            By default no sorting is being done and patches are returned in a row-major order.
+        pad_mode: {``"constant"``, ``"edge"``, ``"linear_ramp"``, ``"maximum"``, ``"mean"``,
+            ``"median"``, ``"minimum"``, ``"reflect"``, ``"symmetric"``, ``"wrap"``, ``"empty"``}
+            One of the listed string values or a user supplied function. Defaults to ``"wrap"``.
+            See also: https://numpy.org/doc/stable/reference/generated/numpy.pad.html
+        pad_opts: padding options, see `numpy.pad`
+        seed: random seed to generate offsets
+
+    """
+
+    backend = [TransformBackends.TORCH, TransformBackends.NUMPY]
+
+    def __init__(
+        self,
+        patch_size: Sequence[int],
+        min_start_pos: Optional[Union[Sequence[int], int]] = None,
+        max_start_pos: Optional[Union[Sequence[int], int]] = None,
+        fix_num_patches: Optional[int] = None,
+        overlap: Union[Sequence[float], float] = 0.0,
+        sort_key: Optional[Union[Callable, str]] = None,
+        pad_mode: Union[NumpyPadMode, str] = NumpyPadMode.CONSTANT,
+        pad_opts: Optional[Dict] = None,
+        seed: int = 0,
```

Comment:
I think we should not set the random seed within the transform.
CC @wyli .

Thanks.

Reply:
I followed this: https://github.com/Project-MONAI/MONAI/blob/eb60b9caf0a103af53ab5f3d620ceb2c116a4851/monai/apps/datasets.py#L81

Reply:
Hi @drbeh ,

Here is the reference design for transforms:
https://github.com/Project-MONAI/MONAI/wiki/Developer-Guide-Transforms#randomized-transforms
We don't need to set `seed` within a transform, users can call `set_random_state(seed=XXX)` outside a random transform if they want deterministic behavior. Otherwise, it's non-deterministic behavior for every call.

Datasets don't have an internal random state, so we explicitly added the `seed` option to control the logic.

CC @ericspod @wyli for more discussion if necessary.

Thanks.

Reply:
That makes sense! I removed seed.